metrics/seqeval/seqeval.py

```diff
@@ -38,19 +40,110 @@
 """
 
 _KWARGS_DESCRIPTION = """
-Produces BLEU scores along with its sufficient statistics
+Produces labelling scores along with its sufficient statistics
 from a source against one or more references.
 
 Args:
     predictions: List of List of predicted labels (Estimated targets as returned by a tagger)
     references: List of List of reference labels (Ground truth (correct) target values)
 Returns:
-    'accuracy': accuracy,
-    'precision': precision,
-    'recall': recall,
-    'f1': F1 score, also known as balanced F-score or F-measure,
+    Overall:
+        'accuracy': accuracy,
+        'precision': precision,
+        'recall': recall,
+        'f1': F1 score, also known as balanced F-score or F-measure,
+    Per type:
+        'precision': precision,
+        'recall': recall,
+        'f1': F1 score, also known as balanced F-score or F-measure,
 """
 
+def end_of_chunk(prev_tag, tag, prev_type, type_):
+    """Checks if a chunk ended between the previous and current word.
+    Args:
+        prev_tag: previous chunk tag.
+        tag: current chunk tag.
+        prev_type: previous type.
+        type_: current type.
+    Returns:
+        chunk_end: boolean.
+    """
+    chunk_end = False
+
+    if prev_tag == 'E': chunk_end = True
+    if prev_tag == 'S': chunk_end = True
+
+    if prev_tag == 'B' and tag == 'B': chunk_end = True
```

Comment:
Could we maybe replace all those if lines with `if prev_tag in ["B", "I"] and tag in ["B", "S", "O"]` or prev_tag in ["E", "S"]? I think this covers all cases

Reply:
Thanks! I'm looking at it!

Reply:
Does-it look better now?

Reply:
great!